apps/server/src/modules/pseudonym/service/pseudonym.service.ts

```diff
@@ -1,14 +1,31 @@
 import { Injectable } from '@nestjs/common';
 import { EntityId, PseudonymDO } from '@shared/domain';
 import { PseudonymsRepo } from '@shared/repo';
+import { v4 as uuidv4 } from 'uuid';
 
 @Injectable()
 export class PseudonymService {
 	constructor(private readonly pseudonymRepo: PseudonymsRepo) {}
 
 	public async findByUserIdAndToolId(userId: EntityId, toolId: EntityId): Promise<PseudonymDO> {
-		const pseudonymPromise: Promise<PseudonymDO> = this.pseudonymRepo.findByUserIdAndToolId(userId, toolId);
+		const pseudonymPromise: Promise<PseudonymDO> = this.pseudonymRepo.findByUserIdAndToolIdOrFail(userId, toolId);
 
 		return pseudonymPromise;
 	}
+
+	public async requestPseudonym(userId: EntityId, toolId: EntityId): Promise<PseudonymDO> {
```

Comment:
The requestPseudonym make more then only requesting. It also create a new one if it is not exists. It is possible that the create can be run into a error, timeout or what ever. In this case a error is created that it can not be saved. It can be confuse if you only see the name "requestPseudonym". I think it make sense to add the second operations with create if not exists also into the name.